builtin/providers/azurerm/resource_arm_local_network_gateway.go

```diff
@@ -0,0 +1,158 @@
+func resourceArmLocalNetworkGatewayCreate(d *schema.ResourceData, meta interface{}) error {
+	lnetClient := meta.(*ArmClient).localNetConnClient
+
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	ipAddress := d.Get("gateway_address").(string)
+
+	// NOTE: due to the including-but-different relationship between the ASM
+	// and ARM APIs, one may set the following local network gateway type to
+	// "Classic" and basically get an old ASM local network connection through
+	// the ARM API. This functionality is redundant with respect to the old
+	// ASM-based implementation which we already have, so we just use the
+	// new Resource Manager APIs here:
+	typ := "Resource Manager"
+
+	// fetch the 'address_space_prefix'es:
+	prefixes := []string{}
+	for _, pref := range d.Get("addres_space").([]interface{}) {
+		prefixes = append(prefixes, pref.(string))
+	}
+
+	// NOTE: result ignored here; review below...
+	resp, err := lnetClient.CreateOrUpdate(resGroup, name, network.LocalNetworkGateway{
+		Name:     &name,
+		Location: &location,
+		Type:     &typ,
+		Properties: &network.LocalNetworkGatewayPropertiesFormat{
+			LocalNetworkAddressSpace: &network.AddressSpace{
+				AddressPrefixes: &prefixes,
+			},
+			GatewayIPAddress: &ipAddress,
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("Error reading the state of Azure ARM Local Network Gateway '%s': %s", name, err)
+	}
+
+	// NOTE: we either call read here or basically repeat the reading process
+	// with the ignored network.LocalNetworkGateway result of the above:
+	d.SetId(*resp.ID)
+	return resourceArmLocalNetworkGatewayRead(d, meta)
+}
+
+// resourceArmLocalNetworkGatewayRead goes ahead and reads the state of the corresponding ARM local network gateway.
+func resourceArmLocalNetworkGatewayRead(d *schema.ResourceData, meta interface{}) error {
+	lnetClient := meta.(*ArmClient).localNetConnClient
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+
+	log.Printf("[INFO] Sending GET request to Azure ARM for local network gateway '%s'.", name)
+	lnet, err := lnetClient.Get(resGroup, name)
+	if lnet.StatusCode == http.StatusNotFound {
+		// it means that the resource has been deleted in the meantime...
+		d.SetId("")
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("Error reading the state of Azure ARM local network gateway '%s': %s", name, err)
+	}
+
+	d.Set("resource_guid", *lnet.Properties.ResourceGUID)
+	d.Set("gateway_address", *lnet.Properties.GatewayIPAddress)
+
+	prefs := []string{}
+	if ps := *lnet.Properties.LocalNetworkAddressSpace.AddressPrefixes; ps != nil {
+		prefs = ps
+	}
+	d.Set("address_space", prefs)
+
+	return nil
+}
+
+// resourceArmLocalNetworkGatewayUpdate goes ahead and updates the corresponding ARM local network gateway.
+func resourceArmLocalNetworkGatewayUpdate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
We can drop this whole function and just use the `resourceArmLocalNetworkGatewayCreate` for the update func in the resource declaration.